tests/vcs/mercurial/HgCommitBearTest.py

```diff
@@ -0,0 +1,504 @@
+        self.assertIn('shortlog_length', metadata.optional_params)
+        self.assertIn('body_line_length', metadata.optional_params)
+        self.assertIn('body_close_issue', metadata.optional_params)
+
+    def test_hg_failure(self):
+        # The only case where hg log gives error is in case its not a repo
+        # so do a log on non-existent repo to perform failure check
+        run_shell_command('mv .hg .hgback')
+        self.assertEqual(self.run_uut(), [])
+        run_shell_command('mv .hgback .hg')
+
+        hg_error = self.msg_queue.get().message
+        self.assertEqual(hg_error[:3], 'hg:')
+
+        self.assert_no_msgs()
+
+    def test_empty_message(self):
+        self.hg_commit('')
+
+        self.assertEqual(self.run_uut(),
+                         ['HEAD commit has no message.'])
+        self.assert_no_msgs()
+
+        self.assertEqual(self.run_uut(allow_empty_commit_message=True),
+                         [])
+        self.assert_no_msgs()
+
+    @unittest.mock.patch('bears.vcs.mercurial.HgCommitBear.run_shell_command',
+                         return_value=('one-liner-message\n', ''))
+    def test_pure_oneliner_message(self, patch):
+        self.assertEqual(self.run_uut(), [])
+        self.assert_no_msgs()
+
+    def test_shortlog_checks_length(self):
+        self.hg_commit('Commit messages that nearly exceed default limit..')
+
+        self.assertEqual(self.run_uut(), [])
+        self.assert_no_msgs()
+
+        self.assertEqual(self.run_uut(shortlog_length=17),
+                         ['Shortlog of the HEAD commit contains 50 '
+                          'character(s). This is 33 character(s) longer than '
+                          'the limit (50 > 17).'])
+        self.assert_no_msgs()
+
+        self.hg_commit('Add a very long shortlog for a bad project history.')
+        self.assertEqual(self.run_uut(),
+                         ['Shortlog of the HEAD commit contains 51 '
+                          'character(s). This is 1 character(s) longer than '
+                          'the limit (51 > 50).'])
+        self.assert_no_msgs()
+
+    def test_shortlog_checks_shortlog_trailing_period(self):
+        self.hg_commit('Shortlog with dot.')
+        self.assertEqual(self.run_uut(shortlog_trailing_period=True), [])
+        self.assertEqual(self.run_uut(shortlog_trailing_period=False),
+                         ['Shortlog of HEAD commit contains a period at end.'])
+        self.assertEqual(self.run_uut(shortlog_trailing_period=None), [])
+
+        self.hg_commit('Shortlog without dot')
+        self.assertEqual(
+            self.run_uut(shortlog_trailing_period=True),
+            ['Shortlog of HEAD commit contains no period at end.'])
+        self.assertEqual(self.run_uut(shortlog_trailing_period=False), [])
+        self.assertEqual(self.run_uut(shortlog_trailing_period=None), [])
+
+    def test_shortlog_wip_check(self):
+        self.hg_commit('[wip] Shortlog')
```

Comment:
out of interest, does 'WIP: Shortlog` also trigger this?

If not, please create an issue about that.

Reply:
Nope, it is not triggered.

Will make an issue :)

Reply:
Done https://github.com/coala/coala-bears/issues/2471